adobe/spectrum-web-components
# fix(toast): Toast should have API for overriding the icon's alt text

Added `label` prop to Toast that allows customization of the `sp-icon`'s label.

## Description

- Previously, the label for the `sp-icon` used in a `sp-toast` was hardcoded to `Success`, `Error` and `Information` for `positive`, `negative` and `info` toasts.
- The addition of the label prop allows one to customize the string. This is useful when we want to use the string `Warning` for a `negative` toast.
- Update storybook to add `label` prop and set value of this prop.

## Related issue(s)
fixes https://github.com/adobe/spectrum-web-components/issues/3406


## Motivation and context

The addition of the label prop allows one to customize the string. This is useful when we want to use the string `Warning` for a `negative` toast.

## How has this been tested?
1. Navigate to storybook locally
2. Go to the Toast component in the sidebar.
5. In DevTools, select the icon with the inspect tool. `sp-icon-x` is in `sp-toast`'s shadow-root.
6. Notice in the `Controls` tab, locate  the new attribute `label`.
4. Type in any string and re-render the toast using `Remount component` in the top bar.
7. Verify that the string typed in step 4 is the present under the `label` prop of `sp-icon-x`. 

## Screenshots (if appropriate)
<img width="900" alt="Screenshot 2024-03-07 at 10 19 37 PM" src="https://github.com/adobe/spectrum-web-components/assets/21351513/a370db28-745f-4a78-bdf4-347156ce3ce4">


## Types of changes


## Checklist


## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
